src/Middleware/RequestQueue/src/SemaphoreWrapper.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.RequestQueue
+{
+    public class SemaphoreWrapper
+    {
+        private static SemaphoreSlim semaphore;
+
+        public SemaphoreWrapper(int qlength)
+        {
+            semaphore = new SemaphoreSlim(qlength);
```

Comment:
`SemaphoreSlim` is `IDisposable`, so we should be disposing it in this wrapper and properly using `using` in our tests (good time to try out the new C# 8 feature: `using var wrapper = new SemaphoreWrapper(...)` ;))